pkg/vcs/vcs.go

```diff
@@ -58,3 +85,117 @@ func VCS(path string) (result v1.VCS) {
 
 	return
 }
+
+func ImageInfoFromApp(ctx context.Context, app *apiv1.App, cloneDir string) (string, string, error) {
+	vcs := app.Status.AppImage.VCS
+	if len(vcs.Remotes) == 0 {
+		return "", "", fmt.Errorf("clone can only be done on an app built from a git repository")
+	}
+	if vcs.Acornfile == "" || vcs.BuildContext == "" {
+		return "", "", fmt.Errorf("app is missing required vcs information, image must be rebuilt with a newer acorn cli")
+	}
+
+	for _, remote := range vcs.Remotes {
+		// Determine the repository name from the repo url
+		idx := strings.LastIndex(remote, "/")
+		if idx < 0 || idx >= len(remote)-1 {
+			fmt.Printf("failed to determine repository name %q\n", remote)
+			continue
+		}
+		workdir := filepath.Join(cloneDir, strings.TrimSuffix(remote[idx+1:], ".git"))
+
+		// Check for the directory we want to use
+		f, err := os.Open(workdir)
+		if err == nil {
+			// Directory exists, check if empty
+			_, err = f.ReadDir(1)
+			if err != nil {
+				// Directory is empty, clone the repo
+				err = gitClone(ctx, workdir, remote)
+				if err != nil {
+					fmt.Printf("%v\n", err)
```

Comment:
Can you customize these error messages a bit to make debugging easier? Like:
"Failed to clone remote %v. Skipping. Error: %v"